source/Preferences.cpp

```diff
@@ -182,6 +185,21 @@ bool Preferences::ZoomViewOut()
 
 
 
+void Preferences::ToggleScreenMode()
+{
+	GameWindow::ToggleFullscreen();
+}
+
+
+
+const string &Preferences::ScreenModeSetting()
+{
+	screenModeIndex = GameWindow::IsFullscreen();
```

Comment:
This should be changed in the function above, when the preference is actually changed.